tools/query-udevadm.py

```diff
@@ -0,0 +1,162 @@
+matching the input pattern.
+
+ex. `python3 query-udevadm.py --pattern=/dev/video*`
+`python3 query-udevadm.py --pattern=/dev/disk/by-id/* --indent=2`
+
+The patterns are matched according to the python glob module. This only provides support 
+for simple unix style patterns.
+
+See also, for example, (or similar):
+`udevadm info /dev/sda | sed -n 's/E: //p'`
+
+The json format is:
+{
+    "devices":[
+            {
+                "path":<pattern match path>
+                "companionDevice": <known/expected devices in companion>
+                "udevInfo":{
+                    <udev-attr>:<udev-value>,
+                    ...
+                }
+            },
+            ... 
+        ]
+    }
+}
+'''
+
+# TODO should we generally move os to subprocess in companion?
+# TODO python2 support
+import sys
+import subprocess
+import glob
+import argparse
+import json
+
+PARSER = argparse.ArgumentParser(description=__doc__, formatter_class=argparse.RawTextHelpFormatter)
+PARSER.add_argument('--pattern',
+                    action="store",
+                    type=str,
+                    default="/dev/serial/by-id/*",
+                    help="Path to search for usb devices."
+                    )
+PARSER.add_argument('--indent',
+                    action="store",
+                    type=int,
+                    default=None,
+                    help="Indent level for json output formatting."
+                    )
+ARGS = PARSER.parse_args()
+
+companionFamiliarDevices = {
+    "Pixhawk Autopilot":
+    {
+        "ID_MODEL" : "PX4_FMU_v2.x",
+        "ID_SERIAL" : "3D_Robotics_PX4_FMU_v2.x_0"
+    },
+    "Pixhawk Autopilot Bootloader":
+    {
+        "ID_MODEL" : 'PX4_BL_FMU_v2.x',
+        "ID_SERIAL" : "3D_Robotics_PX4_BL_FMU_v2.x_0"
+    },
+    "Blue Robotics HD Low Light USB Camera":
+    {
+        "ID_VENDOR_ID" : "05a3",
+        "ID_MODEL_ID" : "9422"
+    },
+    "Raspberry Pi Camera Module":
+    {
+        "ID_V4L_PRODUCT" : "mmal service 16.1",
+        "COLORD_KIND" : "camera"
+    },
+    "Logitech c920 Camera":
+    {
+        "ID_MODEL_ID" : "082d",
+        "ID_VENDOR_ID" : "046d"
+    }
+}
+
+ret = {
+    "devices":[]
+}
+
+def getUdevInfo(devicePath):
+    try:
+        output = subprocess.check_output(
+            ["udevadm", "info", devicePath],
+            stderr = subprocess.DEVNULL, # don't print stderr in output
+            universal_newlines=True) # needed for the output to be a string?
+
+        fields = output.split('\n')
+    except subprocess.CalledProcessError as e:
+        if e.returncode is 4:
+            print("no udevadm info for %s" % devicePath, file=sys.stderr)
+            return {}
+        else:
+            raise e
+
+    ret = {}
+
+    # for each line "E: KEY=VALUE"
+    for field in fields:
+        # remove the first three characters, "E: "
+        field = field[3:]
+
+        kvPair = field.split('=')
+
+        if len(kvPair) > 1:
+            ret[kvPair[0]] = kvPair[1]
```

Comment:
I think that a regex naming group will make it more simple.

Reply:
I think it is already a simple parse, no need to use regex.

Reply:
I added the regex, the code is one or two lines shorter